session-replay/Tests/DatadogSessionReplayTests/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UITextViewRecorderTests.swift

```diff
@@ -0,0 +1,73 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import XCTest
+@testable import DatadogSessionReplay
+
+// swiftlint:disable opening_brace
+class UITextViewRecorderTests: XCTestCase {
+    private let recorder = UITextViewRecorder()
+    /// The label under test.
+    private let textView = UITextView()
+    /// `ViewAttributes` simulating common attributes of text view's `UIView`.
+    private var viewAttributes: ViewAttributes = .mockAny()
+
+    func testWhenTextViewIsNotVisible() throws {
+        // When
+        viewAttributes = .mock(fixture: .invisible)
+
+        // Then
+        let semantics = try XCTUnwrap(recorder.semantics(of: textView, with: viewAttributes, in: .mockAny()))
+        XCTAssertTrue(semantics is InvisibleElement)
+        XCTAssertNil(semantics.wireframesBuilder)
+    }
+
+    func testWhenTextViewHasText() throws {
+        // Given
+        let randomText: String = .mockRandom()
+        textView.textColor = .mockRandom()
+        textView.layoutManager.allowsNonContiguousLayout = true
```

Comment:
Without this it fails on enigmatic:
```
testWhenTextViewHasText(): CALayer position contains NaN: [0 nan]. Layer: <CALayer:0x6000030319a0; position = CGPoint (0 15); bounds = CGRect (0 0; 0 30); delegate = <<_UITextContainerView: 0x104f0d9e0; frame = (0 0; 0 30); backgroundColor = UIExtendedGrayColorSpace 0 0; layer = <CALayer: 0x6000030319a0>> minSize = {0, 0}, maxSize = {1.7976931348623157e+308, 1.7976931348623157e+308}, textContainer = <NSTextContainer: 0x600000110640 size = (0.000000,inf); widthTracksTextView = YES; heightTracksTextView = NO>; exclusionPaths = 0x1babfefa8; lineBreakMode = 0>; sublayers = (<CALayer: 0x600003030dc0>); allowsGroupOpacity = YES; backgroundColor = <CGColor 0x600001302580> [<CGColorSpace 0x600001406880> (kCGColorSpaceICCBased; kCGColorSpaceModelMonochrome; Generic Gray Gamma 2.2 Profile; extended range)] ( 0 0 )> (CALayerInvalidGeometry)
```

Reply:
How about explaining this in code comment next to it? Otherwise it will be very unclear and will require `git blame` to find the original purpose. Typically we do it somehow this:

```suggestion
        // RUMM-2681 Following is required to avoid "CALayer position contains NaN: [0 nan]. (...) (CALayerInvalidGeometry)" error
        textView.layoutManager.allowsNonContiguousLayout = true
```